bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -369,7 +391,30 @@ public BookKeeper(ClientConfiguration conf, ZooKeeper zk)
      */
     public BookKeeper(ClientConfiguration conf, ZooKeeper zk, EventLoopGroup eventLoopGroup)
             throws IOException, InterruptedException, BKException {
-        this(conf, validateZooKeeper(zk), validateEventLoopGroup(eventLoopGroup), NullStatsLogger.INSTANCE,
+        this(conf, validateZooKeeper(zk), validateEventLoopGroup(eventLoopGroup), null);
+    }
+
+    /**
+     * Create a bookkeeper client but use the passed in zookeeper client and
+     * client event loop group instead of instantiating those.
+     *
+     * @param conf
+     *          Client Configuration Object
+     *          {@link ClientConfiguration}
+     * @param zk
+     *          Zookeeper client instance connected to the zookeeper with which
+     *          the bookies have registered. The ZooKeeper client must be connected
+     *          before it is passed to BookKeeper. Otherwise a KeeperException is thrown.
+     * @param eventLoopGroup
+     *          An event loop group that will be used to create connections to the bookies
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws BKException in the event of a bookkeeper connection error
+     */
+    public BookKeeper(ClientConfiguration conf, ZooKeeper zk, EventLoopGroup eventLoopGroup,
```

Comment:
Please don't create another public constructor for BookKeeper. The client should be constructed through the builder.

Reply:
Ok, I was doing this since this is the constructor we use in Pulsar :)